File: OODPFP/src/main/java/command/runner/RemoteLoader.java
package command.runner;

import command.Command;
import command.MacroCommand;
import command.appliances.Hottub;
import command.appliances.Light;
import command.appliances.Stereo;
import command.appliances.TV;
import command.commands.*;

public class RemoteLoader {

    public static void main(String[] args) {

        RemoteControl remoteControl = new RemoteControl();

        Light light = new Light("Living Room");
        TV tv = new TV("Living Room");
        Stereo stereo = new Stereo("Living Room");
        Hottub hottub = new Hottub();

        Command[] partyOn = {
                new LightOnCommand(light), new StereoOnCommand(stereo),
                new TVOnCommand(tv), new HottubOnCommand(hottub)};

        Command[] partyOff = {new LightOffCommand(light), new StereoOffCommand(stereo),
                new TVOffCommand(tv), new HottubOffCommand(hottub)};

        remoteControl.setCommand(0, new MacroCommand(partyOn), new MacroCommand(partyOff));

        System.out.println(remoteControl);
        System.out.println("--- Pushing Macro On---");
        remoteControl.onButtonWasPushed(0);
        System.out.println("--- Pushing Macro Off---");
        remoteControl.offButtonWasPushed(0);
    }
}

File: OODPFP/src/main/scala/commandFP/RemoteLoader.scala
package commandFP

import commandFP.Commands._
import commandFP.appliances.{Hottub, Light, Stereo, TV}

object RemoteLoader {
  def main(args: Array[String]) {
    val light = Light("living room")
    val tv = TV("living room")
    val stereo = Stereo("living room")
    val hottub = Hottub()

    val on: Command = macroCommand(light.on,
      stereo.on, tvOn(tv), hottubOn(hottub))

    val off: Command = macroCommand(light.off,
      stereo.off, tv.off, hottubOff(hottub))

    val remoteControl = RemoteControl(Seq(on), Seq(off))

    println("--- Pushing Macro On---")
    remoteControl.pushOnButton(0)
    println("--- Pushing Macro Off---")
    remoteControl.pushOffButton(0)
  }
}
